### PR TITLE
:ambulance: possible multiple multisessions with same multisession ID

### PR DESCRIPTION
**Pull request by KolushovAlexandr**

### CODE REVIEW

**Review by yelizariev — COMMENTED** [review]
You will need to update version

**Review by Ramil-Mukhametzyanov — APPROVED** [review]

File: pos_multi_session_sync/controllers/pos_multi_session.py [hunk 1]
```diff
@@ -32,13 +32,18 @@ def multi_session_update(self, multi_session_id, message, dbname, user_ID):
...
-                             ('dbname', '=', dbname)])
-        if not ms:
-            ms = ms_model.create({'multi_session_ID': int(multi_session_id), 'dbname': dbname})
-        ms = ms.with_context(user_ID=user_ID, phantomtest=phantomtest)
+        multi_session = ms_model.search([('multi_session_ID', '=', int(multi_session_id)),
+                                         ('dbname', '=', dbname)])
+        if len(multi_session) > 1:
+            multi_session.unlink()
+        elif len(multi_session) == 1 and multi_session.dbname != dbname:
+            multi_session = ms_model.write({'dbname': dbname})
```
Comment (yelizariev) [inline pos_multi_session_sync/controllers/pos_multi_session.py]: wrong code and Impossible case ( multi_session.dbname is always dbname, ms_model is not a recordset

**Review by yelizariev — CHANGES_REQUESTED** [review]

File: pos_multi_session_sync/controllers/pos_multi_session.py [hunk 2]
```diff
@@ -32,13 +32,18 @@ def multi_session_update(self, multi_session_id, message, dbname, user_ID):
...
             ms_model = ms_model.sudo()
-        ms = ms_model.search([('multi_session_ID', '=', int(multi_session_id)),
-                             ('dbname', '=', dbname)])
-        if not ms:
-            ms = ms_model.create({'multi_session_ID': int(multi_session_id), 'dbname': dbname})
-        ms = ms.with_context(user_ID=user_ID, phantomtest=phantomtest)
+        multi_session = ms_model.search([('multi_session_ID', '=', int(multi_session_id)),
+                                         ('dbname', '=', dbname)])
+        if len(multi_session) > 1:
+            multi_session.unlink()
```
Comment (yelizariev) [inline pos_multi_session_sync/controllers/pos_multi_session.py:40]: Any idea how it's possible? Anyway, you need to add a comment about this strange case

File: pos_multi_session_sync/controllers/pos_multi_session.py [hunk 3]
```diff
@@ -32,13 +32,18 @@ def multi_session_update(self, multi_session_id, message, dbname, user_ID):
...
-                             ('dbname', '=', dbname)])
-        if not ms:
-            ms = ms_model.create({'multi_session_ID': int(multi_session_id), 'dbname': dbname})
-        ms = ms.with_context(user_ID=user_ID, phantomtest=phantomtest)
+        multi_session = ms_model.search([('multi_session_ID', '=', int(multi_session_id))])
+        if len(multi_session) > 1:
+            # somehow this case happened, but there is no scenario to reproduce
+            multi_session.unlink()
+        elif len(multi_session) == 1 and multi_session.dbname != dbname:
+            multi_session = multi_session.write({'dbname': dbname})
```
Comment (yelizariev) [inline pos_multi_session_sync/controllers/pos_multi_session.py]: Why?

**Review by yelizariev — APPROVED** [review]

**Pull request merged**